flowmachine/flowmachine/core/hour_slice.py

```diff
@@ -0,0 +1,258 @@
+                raise ValueError(
+                    f"Invalid hour value: {hour} (must be between 0 and 23, inclusively)"
+                )
+            if not (0 <= minutes and minutes < 60):
+                raise ValueError(
+                    f"Invalid minutes value: {minutes} (must be between 0 and 59, inclusively)"
+                )
+        else:
+            raise ValueError(
+                f"Input argument must be a string the format 'HH:MM'. Got: {hour_str}"
+            )
+
+    def filter_timestamp_column(self, ts_col, cmp_op):
+        """
+        Filter timestamp column by comparing to this hour-of-day using the given comparison operator.
+
+        Parameters
+        ----------
+        ts_col : sqlalchemy column
+            The timestamp column to filter.
+        cmp_op : callable
+            Comparison operator to use. For example: `operator.lt`, `operator.ge`.
+
+        Returns
+        -------
+        sqlalchemy.sql.elements.BooleanClauseList
+            Sqlalchemy expression representing the filtered timestamp column.
+            This can be used in WHERE clauses of other sql queries.
+        """
+        if self.hour_str is None:
+            # no filtering needed
+            return true()
+        else:
+            return cmp_op(func.to_char(ts_col, "HH24:MI"), self.hour_str)
+
+
+class HourInterval:
+    """
+    Represents an interval of hours during the day which is repeated
+    regularly, for example each day, or every Tuesday. The interval is
+    considered to be half-open, i.e. the left endpoint is included and
+    the right endpoint is excluded. For example: 08:30 <= HH:MM < 16:00.
+
+    Parameters
+    ----------
+    start_hour : str
+        Start hour of this hour interval in the format 'HH:MM' (e.g. '08:00').
+    stop_hour : str (optional)
+        Stop hour of this hour interval in the format 'HH:MM' (e.g. '19:30').
+        The stop hour can also be `None`, indicating that the hour interval
+        extends until midnight.
+    freq : str
+        Frequency at which the underlying time interval is repeated. This
+        must be either "day" or "week". In the latter case the `weekday`
+        argument must also be provided.
+    weekday : str
+        The day of the week for which this hour interval is valid. This argument
+        is only relevant if `freq="week"` and is ignored otherwise.
+    """
+
+    def __init__(
+        self, *, start_hour: str, stop_hour: str, freq: str, weekday: str = None
+    ):
+        self.start_hour = HourOfDay(start_hour)
+        self.stop_hour = HourOfDay(stop_hour)
+        self.period = make_hour_interval_period(freq, weekday=weekday)
+
+    def __repr__(self):
+        return (
+            f"HourInterval(start_hour={self.start_hour!r}, stop_hour={self.stop_hour!r}, "
+            f"freq={self.period.freq!r}, weekday={self.period.weekday!r})"
+        )
+
+    def filter_timestamp_column(self, ts_col):
```

Comment:
Type annotations for this, given the `__init__` is annotated?

Reply:
Yes, I'd like to add type annotations, but I'm not entirely sure what the most general valid class for the timestamp column is. Currently when I run
```
from flowmachine.core.sqlalchemy_table_definitions import EventsCallsTable 
print(type(EventsCallsTable.datetime))
```                                                                                                                              
it returns `<class 'sqlalchemy.orm.attributes.InstrumentedAttribute'>`, but it feels to me this is a fairly specific class and the method would accept more general ones. But I can add this particular one for the time being if you think that's ok?